src/team3player/RobotPlayer.java

```diff
@@ -232,3 +260,12 @@ static void tryBlockchain() throws GameActionException {
         // System.out.println(rc.getRoundMessages(turnCount-1));
     }
 }
+class MemoryForHQ {
+     int numOfMiners = 0;
+}
+class MemoryforMiner {
+     MapLocation HQ = null;
+}
+class MemoryForRefinery {
+
+}
```

Comment:
You may want to make these nested classes within the player.

Reply:
I tried that out and it didn't build because the memory object are not static. If the classes were nested objects of their type would have to be static. The problem with that is they would revseve the memory necessary for each object even if we didn't want to use that bot. I recommend we keep them outside the robotplayer class